From ab88feb668e3462ee235726fcedd24eee6d1debf Mon Sep 17 00:00:00 2001
From: Gatien Chevallier <gatien.chevallier@foss.st.com>
Date: Mon, 18 Mar 2024 10:24:17 +0100
Subject: [PATCH 0698/1695] dts: stm32: prepare use of GPIO instead of pinctrl
 for external tampers

This change adds "st,tamp-id" and "tamper-gpios" properties to tamper
sub nodes to prepare the use of GPIO instead of pinctrl for tampers.

Upstream-Status: Inappropriate [ST downstream https://github.com/STMicroelectronics/optee_os.git b7a3173ee6e61659b25a05196696dc71c8f8ad61 dts: stm32: prepare use of GPIO instead of pinctrl for external tampers]
Signed-off-by: Gatien Chevallier <gatien.chevallier@foss.st.com>
Change-Id: I7c876332a236d1f59ff23a698eb635d183a4a8ab
Reviewed-on: https://gerrit.st.com/c/mpu/oe/optee/optee_os/+/367345
Domain-Review: Yann GAUTIER <yann.gautier@foss.st.com>
Tested-by: Gatien CHEVALLIER <gatien.chevallier@st.com>
Reviewed-by: Gatien CHEVALLIER <gatien.chevallier@st.com>
ACI: CITOOLS <MDG-smet-aci-reviews@list.st.com>
ACI: CIBUILD <MDG-smet-aci-builds@list.st.com>
---
 core/arch/arm/dts/stm32mp135f-dk.dts  | 4 ++++
 core/arch/arm/dts/stm32mp235f-dk.dts  | 2 ++
 core/arch/arm/dts/stm32mp257f-dk.dts  | 2 ++
 core/arch/arm/dts/stm32mp257f-ev1.dts | 2 ++
 4 files changed, 10 insertions(+)

diff --git a/core/arch/arm/dts/stm32mp135f-dk.dts b/core/arch/arm/dts/stm32mp135f-dk.dts
index 3cfa364a2..49dd6b484 100644
--- a/core/arch/arm/dts/stm32mp135f-dk.dts
+++ b/core/arch/arm/dts/stm32mp135f-dk.dts
@@ -642,12 +642,16 @@
 		status = "okay";
 		reg = <EXT_TAMPER_2 0>;
 		pinctrl-0 = <&tamp0_in2_pin_a>;
+		tamper-gpios = <&gpioa 6 0>;
 		st,tamp-mode = <TAMPER_CONFIRMED_MODE>;
+		st,tamp-id = <2>;
 	};
 
 	/* Connect pin8 and pin22 from CN8 */
 	tamp-active@1 {
 		pinctrl-0 = <&tamp0_in3_pin_a &tamp0_out1_pin_a>;
+		tamper-gpios = <&gpioc 0 0>, <&gpioi 0 0>;
+		st,tamp-id = <3>, <1>;
 		status = "disabled";
 	};
 };
diff --git a/core/arch/arm/dts/stm32mp235f-dk.dts b/core/arch/arm/dts/stm32mp235f-dk.dts
index 9bc1d73f8..8b4150ef1 100644
--- a/core/arch/arm/dts/stm32mp235f-dk.dts
+++ b/core/arch/arm/dts/stm32mp235f-dk.dts
@@ -452,7 +452,9 @@
 	tamp_button: tamp-passive@3 {
 		status = "disabled";
 		reg = <EXT_TAMPER_3 0>;
+		tamper-gpios = <&gpioz 2 0>;
 		st,tamp-mode = <TAMPER_CONFIRMED_MODE>;
+		st,tamp-id = <3>;
 	};
 };
 
diff --git a/core/arch/arm/dts/stm32mp257f-dk.dts b/core/arch/arm/dts/stm32mp257f-dk.dts
index 0cfc192cc..947805441 100644
--- a/core/arch/arm/dts/stm32mp257f-dk.dts
+++ b/core/arch/arm/dts/stm32mp257f-dk.dts
@@ -440,7 +440,9 @@
 	tamp_button: tamp-passive@3 {
 		status = "disabled";
 		reg = <EXT_TAMPER_3 0>;
+		tamper-gpios = <&gpioz 2 0>;
 		st,tamp-mode = <TAMPER_CONFIRMED_MODE>;
+		st,tamp-id = <3>;
 	};
 };
 
diff --git a/core/arch/arm/dts/stm32mp257f-ev1.dts b/core/arch/arm/dts/stm32mp257f-ev1.dts
index cc7d6f6b8..25fcb0455 100644
--- a/core/arch/arm/dts/stm32mp257f-ev1.dts
+++ b/core/arch/arm/dts/stm32mp257f-ev1.dts
@@ -467,7 +467,9 @@
 	tamp_button: tamp-passive@1 {
 		status = "disabled";
 		reg = <EXT_TAMPER_1 0>;
+		tamper-gpios = <&gpioi 8 0>;
 		st,tamp-mode = <TAMPER_CONFIRMED_MODE>;
+		st,tamp-id = <1>;
 	};
 };
 
-- 
2.47.2

